variables-bastion.tf

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+variable "create_bastion" {
+  default     = true
+  description = "Whether to create a bastion host."
+  type        = bool
+}
+
+variable "bastion_public_ip" {
+  default     = null
+  description = "The IP address of an existing bastion host, if create_bastion = false."
+  type        = string
+}
+
+variable "bastion_allowed_cidrs" {
+  default     = ["0.0.0.0/0"]
+  description = "A list of CIDR blocks to allow SSH access to the bastion host."
+  type        = list(string)
+}
+
+variable "bastion_availability_domain" {
+  default     = null
+  description = "The availability domain for bastion placement. Defaults to first available."
+  type        = string
+}
+
+variable "bastion_nsg_ids" {
+  description = "An additional list of network security group (NSG) IDs for bastion security."
+  default     = []
+  type        = list(string)
+}
+
+variable "bastion_user" {
+  default     = "opc"
+  description = "User for SSH access through bastion host."
+  type        = string
+}
+
+variable "bastion_image_id" {
+  default     = null
+  description = "Image ID for created bastion instance."
+  type        = string
+}
+
+variable "bastion_image_type" {
+  default     = "platform"
+  description = "Whether to use a platform or custom image for the created bastion instance. When custom is set, the bastion_image_id must be specified."
+  type        = string
+  validation {
+    condition     = contains(["custom", "platform"], var.bastion_image_type)
+    error_message = "Accepted values are custom or platform"
+  }
+}
+
+variable "bastion_image_os" {
+  default     = "Oracle Linux"
```

Comment:
Add validation. Value can be Oracle Linux, Ubuntu, SUSE, AlmaLinux, Rocky Linux

Reply:
Validation exists in the form of a precondition on matched `image_id` - the bastion in particular is pretty flexible as it's just used for SSH. If no image is matched, the precondition will yield an error that calls out this value.